pkg/squash/vfs_fs.go

```diff
@@ -0,0 +1,210 @@
+package squash
```

Comment:
So... initially, I was looking at this as "hah, funny name for something that isn't a VFS." As I read it, I realize that I was wrong, and this _is_ a VFS implementation.

VFS implementations suck to read if you haven't read them before. This file, as a whole, needs to be commented a lot more, so that newcomers can have a fighting chance.